### PR TITLE
Getting started with Nodejs EJS templating engine

### LINKED ISSUE #1834: Getting started with Nodejs EJS templating engine
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Introduction paragraph (2-3 paragraphs):
EJS is a simple templating language that lets you generate HTML markup with plain JavaScript. EJS simply stands for Embedded JavaScript templates, and we can use it both server-side or client-side.

In the article I will introduce the reader to EJS templating engine, explain how to add it to a Nodejs project, and finally use it to generate an HTML page. Throughout the tutorial, I will be explaining how to build a webpage using EJS templating engine.

**Basic syntax**
- <% 'Scriptlet' tag, for control-flow, no output.
- <%= Outputs the value into the template (HTML escaped).
- <%- Outputs the unescaped value into the template.

### Key takeaways: 
1. The reader will be able to setup EJS in a Nodejs project.
2. The reader will be able to use various EJS templating tags. 

### References: 
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by qawuor**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1834

### CODE REVIEW

File: articles/nodejs-ejs/index.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+- `const app = express();` creates an instance of the express, making our application an express application.
+- `app.set('views', path.join(__dirname, 'views'));` sets the view directory to `views`. This is where all our `EJS` files will be stored.
+- `app.set('view engine', 'ejs');` sets the view engine to `EJS`.
+- `app.listen(3000)` starts the development server.
+
+start the server by executing the command below.
+```bash
+$ node app.js
+```
+When we navigate to `localhost:3000/` in our browser, a page with all the posts is displayed.
```
Comment (paulodhiambo) [inline articles/nodejs-ejs/index.md:202]: Consider changing "When we navigate to `localhost:3000/` in our browser, a page with all the posts is displayed." to "When we navigate to `localhost:3000/` in our browser, a page with all the posts is displayed as shown below."

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@qawuor amazing tutorial. Review the change and ping me when done.

**Comment by qawuor** [comment]
@paulodhiambo I have made the requested change.

**Review by paulodhiambo — APPROVED** [review]

**Comment by paulodhiambo** [comment]
@qawuor amazing article. @hectorkambow you can give it a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**